MekHQ/unittests/mekhq/campaign/market/ContractMarketAtBGenerationTests.java

```diff
@@ -0,0 +1,1741 @@
+        doReturn(enemyFaction).when(factions).getFaction(eq(enemy));
+
+        Faction pirates = mock(Faction.class);
+        doReturn(pirates).when(factions).getFaction(eq("PIR"));
+
+        Faction rebels = mock(Faction.class);
+        doReturn(rebels).when(factions).getFaction(eq("REB"));
+
+        Systems systems = mock(Systems.class);
+        Systems.setInstance(systems);
+
+        String current = "CURRENT";
+        PlanetarySystem currentSystem = mock(PlanetarySystem.class);
+        when(currentSystem.getId()).thenReturn(current);
+        when(campaign.getCurrentSystem()).thenReturn(currentSystem);
+        doReturn(currentSystem).when(systems).getSystemById(eq(current));
+        doReturn(currentSystem).when(campaign).getSystemByName(eq(current));
+
+        CurrentLocation currentLocation = mock(CurrentLocation.class);
+        when(campaign.getLocation()).thenReturn(currentLocation);
+
+        String missionTarget = "TARGET";
+        PlanetarySystem targetSystem = mock(PlanetarySystem.class);
+        when(targetSystem.getId()).thenReturn(missionTarget);
+        doReturn(targetSystem).when(systems).getSystemById(eq(missionTarget));
+        doReturn(targetSystem).when(campaign).getSystemByName(eq(missionTarget));
+
+        RandomFactionGenerator rfg = mock(RandomFactionGenerator.class);
+        RandomFactionGenerator.setInstance(rfg);
+        when(rfg.getEmployer()).thenReturn(employer);
+        doReturn(enemy).when(rfg).getEnemy(eq(employer), anyBoolean());
+        doReturn(missionTarget).when(rfg).getMissionTarget(anyString(), anyString());
+
+        FactionHints hints = mock(FactionHints.class);
+        doReturn(true).when(hints).isISMajorPower(eq(employerFaction));
+        doReturn(true).when(hints).isISMajorPower(eq(enemyFaction));
+        doReturn(false).when(hints).isNeutral(eq(employerFaction));
+        doReturn(false).when(hints).isNeutral(eq(enemyFaction));
+        when(rfg.getFactionHints()).thenReturn(hints);
+
+        JumpPath jumpPath = mock(JumpPath.class);
+        when(jumpPath.getJumps()).thenReturn(1);
+        doReturn(10.0).when(jumpPath).getTotalTime(any(), anyDouble());
+        doReturn(jumpPath).when(campaign).calculateJumpPath(eq(currentSystem), eq(targetSystem));
+        doReturn(Money.of(1)).when(campaign).calculateCostPerJump(anyBoolean(), anyBoolean());
+
+        ContractMarket market = new ContractMarket();
+
+        AtBContract contract = market.addAtBContract(campaign);
+        assertNotNull(contract);
+    }
+
+    @Test
+    public void addMercWithoutRetainerMinorPowerAtBContractSucceeds() {
+        Campaign campaign = mock(Campaign.class);
+        when(campaign.getFactionCode()).thenReturn("MERC");
+        when(campaign.getRetainerEmployerCode()).thenReturn(null);
+        when(campaign.getUnitRatingMod()).thenReturn(unitRating);
+        when(campaign.getLocalDate()).thenReturn(LocalDate.of(gameYear, 1, 1));
```

Comment:
```suggestion
        when(campaign.getLocalDate()).thenReturn(LocalDate.ofYearDay(gameYear, 1));
```

Reply:
This is the neater one to use for first of the year, and should be done elsewhere